### PR TITLE
fix: parsing of `ssoEnabled` in state.php

### PR DESCRIPTION
**Pull request by pujitm**
read `ssoSubIds` in state.php from `api.json`

## Summary by CodeRabbit

* **New Features**
  * Added a new query to check if Single Sign-On (SSO) is enabled.
  * Updated UI components to dynamically reflect SSO availability via live data.
* **Refactor**
  * Streamlined internal handling of SSO status detection for improved reliability and maintainability.
* **Tests**
  * Enhanced tests for SSO button behavior with mocked live data and added edge case coverage for SSO callback handling.

### CODE REVIEW

File: plugin/source/dynamix.unraid.net/usr/local/emhttp/plugins/dynamix.my.servers/include/api-config.php [hunk 1]
```diff
@@ -95,3 +95,18 @@ public static function getApiVersion()
...
+    public static function getConfig() {
+        $config = json_decode(file_get_contents(self::CONFIG_PATH), true);
+        return $config;
+    }
+
+    public static function isSSOEnabled() {
+        $config = self::getConfig();
+        return !empty($config['ssoSubIds'] ?? '');
+    }
+}
```
[the remark this answers is not in the document]
Comment (pujitm) [reply plugin/source/dynamix.unraid.net/usr/local/emhttp/plugins/dynamix.my.servers/include/api-config.php]: is there a more concise/idiomatic way to return an empty array in all these cases?

**Comment by elibosley** [comment]
Simple I like it
